### PR TITLE
3djc/x7: LS fix limits

### PR DESCRIPTION
**Pull request by 3djc**
Fix an X7 bug where logical switch value where displayed wrong for some types of fields.

While I was at it,  I changed to limits values from static values to getMaximumValue() to match the larger Taranis behavior.

I tested on X7, but it is very possible that this breaks other 128x64 (don't know enough to tell)

ps : my thanks to @kilrah for his ongoing support :)

### CODE REVIEW

**Comment by bsongis** [comment]
In this file there is an `#if defined(CPUARM)` duplicated test, is it possible you remove it? Thanks

**Comment by bsongis** [comment]
Even 2 times!

**Comment by 3djc** [comment]
I really don't have a good tool for that kind of things I found 2, but maybe not the ones you where expecting !

File: radio/src/gui/128x64/model_logical_switches.cpp [hunk 1]
```diff
@@ -509,6 +516,12 @@ void menuModelLogicalSwitches(event_t event)
         case LS_FIELD_V3:
           cs->v3 = CHECK_INCDEC_PARAM(event, cs->v3, v3_min, v3_max);
           break;
+        case LS_FIELD_DURATION:
```
Comment (bsongis) [inline radio/src/gui/128x64/model_logical_switches.cpp]: I don't think it's a good idea to move the case order, it's better to keep the order on the LCD

File: radio/src/gui/128x64/model_logical_switches.cpp [hunk 2]
```diff
@@ -162,11 +159,7 @@ void menuModelLogicalSwitchOne(event_t event)
         else {
 #if defined(TELEMETRY_FRSKY)
           if (v1_val >= MIXSRC_FIRST_TELEM) {
-#if defined(CPUARM)
```
Comment (bsongis) [inline radio/src/gui/128x64/model_logical_switches.cpp]: yes that's what I meant, thanks!

**Pull request merged**